Automattic/jetpack
# Add/waf fetch rules

When WAF rules are generating, the rules will now be provided by the wpcom API. The wpcom API endpoint requires automattician authentication currently. When activating the Jetpack plugin, that information is not yet established. As such, the activation hook was removed from actions.php, and is now is activated exclusively by a CLI command. In the future, activation will also be possibly through a WAF UI.




#### Changes proposed in this Pull Request:

* A Jetpack activation hook has been removed
* WAF CLI commands have been improved for setup and teardown
* Generating the rules now fetches rules from the wpcom API

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:





**TEST 1:** Test the WAF CLI setup command properly generates the rules file. Assumes the site is a fresh Docker site.
1. Checkout this branch
2. Open the site using Jurassic Tube
3. Activate Jetpack and establish connection using your a11n wpcom account
4. Erase rules.php file contents or delete file if it exists
5. Open a docker terminal for the wordpress site
6. Execute `wp jetpack-waf setup normal --user=1 --allow-root` ( note: assumes that the admin account used to set up Jetpack is ID 1 )

**Test 1 Expectation:** The rules.php file should now exist and contain the content sent from the API. Also, the terminal should report two successes:
> Success: Jetpack WAF mode switched to "normal".
> Success: Jetpack WAF has successfully been setup.

**TEST 2:** Test the WAF CLI teardown command is successful. Assumes performing this test following **Test 1**.
1. From the docker terminal, execute `wp jetpack-waf teardown --allow-root`

**Test 2 Expectation:** The rules.php file should now be containing only `<?php`. Also the terminal should report success:
> Success: Jetpack WAF has been deactivated.
